packages/libs/react-ui/src/utils/useTheme/index.ts

```diff
@@ -0,0 +1,133 @@
+'use client';
+import { useCallback, useEffect, useState } from 'react';
+import { darkThemeClass } from '../../styles';
+import type { ITheme } from './utils/constants';
+import { MEDIA, defaultTheme, isServer, storageKey } from './utils/constants';
+import { getSystemTheme } from './utils/getSystemTheme';
+
+interface IUseThemeProps {
+  /**
+   * lockedTheme will lock the theme for the whole application.
+   * It will overwrite the localstorage and even the switching of themes in your system will not update the theme
+   * IMPORTANT: using the useTheme twice in a application can lead to unexpected race conditions
+   * You would need to lock the theme on all the uses of this hook with the same ITheme
+   */
+  lockedTheme?: ITheme;
+  /**
+   * overwriteTheme can be used when you want 1 particular component to have a fixed theme
+   * IMPORTANT: this will also overwrite the lockedTheme for that component
+   */
+  overwriteTheme?: ITheme;
+}
+interface IUseThemeReturnProps {
+  theme: ITheme | undefined;
+  setTheme: (value: ITheme) => void;
+}
+
+const getTheme = (key: string) => {
+  if (isServer) return undefined;
+  let theme: ITheme | undefined = undefined;
+  try {
+    theme = (localStorage.getItem(key) as ITheme) || undefined;
+  } catch (e) {
+    console.error('localStorage is not suported in this browser');
+  }
+  return theme || defaultTheme;
+};
+
+export const useTheme = ({
+  overwriteTheme,
+  lockedTheme,
+}: IUseThemeProps = {}): IUseThemeReturnProps => {
+  const [theme, setThemeState] = useState(() =>
+    lockedTheme ? lockedTheme : getTheme(storageKey),
+  );
+
+  const applyTheme = useCallback((innerTheme: ITheme) => {
+    let resolved = innerTheme;
+    if (!resolved) return;
+
+    // If theme is system, resolve it before setting theme
+    if (innerTheme === 'system') {
+      resolved = getSystemTheme();
+    }
+    if (lockedTheme) {
+      resolved = lockedTheme;
+    }
+
+    const name = resolved;
+    const d = document.documentElement;
```

Comment:
Can we name this one documentElement? I like what you did the line above to rename a variable to match the context but having a single letter variable name the line below kind of undo's the good thing.